api/v1beta2/awsmachine_types.go

```diff
@@ -220,6 +220,7 @@ type AWSMachineSpec struct {
 	//  If this value is selected, CapacityReservationID must be specified to identify the target reservation.
 	// If marketType is not specified and spotMarketOptions is provided, the marketType defaults to "Spot".
 	// +optional
+	// +kubebuilder:validation:XValidation:rule="!has(self.capacityReservationID) || !has(self.marketType) || self.marketType != 'Spot'",message="capacityReservationID may not be set when marketType is Spot"
```

Comment:
This must be added to the parent struct, and not this field, as "self" in the rule will apply to the field if you apply it here, or the parent struct if you apply it to the parent struct